### DISCUSSION TITLE
Failed to open the UI of zipkin

### DISCUSSION THREAD

**Discussion by Yangfisher1** [apache/skywalking#11116] [Q&A] [answered]
Hi,

I'm using `helm` to install skywalking on my k8s cluster as below.
```bash
SKYWALKING_RELEASE_VERSION=4.4.0
SKYWALKING_RELEASE_NAME=skywalking
SKYWALKING_RELEASE_NAMESPACE=monitoring

sudo helm install "${SKYWALKING_RELEASE_NAME}" \
  oci://registry-1.docker.io/apache/skywalking-helm \
  --version "${SKYWALKING_RELEASE_VERSION}" \
  -n "${SKYWALKING_RELEASE_NAMESPACE}" \
  --set oap.image.tag=9.5.0 \
  --set oap.storageType=elasticsearch \
  --set ui.image.tag=9.5.0 \
  --set oap.env.SW_OTEL_RECEIVER=default \
  --set oap.env.SW_RECEIVER_ZIPKIN=default \
  --set oap.ports.zipkin=9411 \
  --set oap.env.SW_OTEL_RECEIVER_ENABLED_OC_RULES="k8s-cluster\,k8s-node\,k8s-service"

```

After that I using port-forward to open the UI of skywalking and it works well.

![image](https://github.com/apache/skywalking/assets/55266940/65f8ff59-b533-4458-a7ef-894d46206e87)

However, when I enter the page `127.0.0.1/zipkin` it shows nothing and throw an error
`Failed to load resource: the server responded with a status of 500 (Internal Server Error)` which indicates that it can't find the `config.json` file of zipkin UI.

![image](https://github.com/apache/skywalking/assets/55266940/a192ddb9-86f5-4382-8b5e-3f44b7949b54)

Is there any configuration problem leading to this failure?

**Comment by kezhenxu94** [comment]
I think you need to set env var `SW_QUERY_ZIPKIN` to `default`

**Reply by Yangfisher1** [reply]
Ok, I tried it by adding `--set oap.env.SW_QUERY_ZIPKIN=default` but still got the same error.

**Reply by kezhenxu94** [reply]
Then please check the Pod logs

**Reply by Yangfisher1** [reply]
It throw an error
```
finishConnect(..) failed: Connection refused: 127.0.0.1/127.0.0.1:9412
```

So is there anything I need to configure more?

**Reply by Yangfisher1** [reply]
The whole request log is below.

```
2023-07-20 06:42:19,403 com.linecorp.armeria.client.logging.LoggingClient 155 [armeria-common-worker-epoll-2-26]
WARN  [] - [creqId=1b8fed63, sreqId=092a9d5f][http://127.0.0.1:9412/zipkin/config.json#GET] 
Request: {startTime=2023-07-20T06:42:19.398Z(1689835339398000), length=0B, duration=3208µs(3208555ns), 
cause=com.linecorp.armeria.client.UnprocessedRequestException: io.netty.channel.AbstractChannel
$AnnotatedConnectException: finishConnect(..) failed: Connection refused: 127.0.0.1/127.0.0.1:9412, 
scheme=none+http, name=GET, headers=[:method=GET, :scheme=http, :path=/zipkin/config.json, host=127.0.0.1:8080, 
sec-ch-ua="Not.A/Brand";v="8", "Chromium";v="114", "Google Chrome";v="114", sec-purpose=prefetch;prerender, 
sec-ch-ua-mobile=?0, user-agent=Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) 
Chrome/114.0.0.0 Safari/537.36, sec-ch-ua-platform="Linux", accept=*/*, purpose=prefetch, sec-fetch-site=same-origin, 
sec-fetch-mode=cors, sec-fetch-dest=empty, referer=http://127.0.0.1:8080/zipkin/, 
accept-encoding=gzip, deflate, br, accept-language=zh-CN,zh;q=0.9]}
```

**Reply by Yangfisher1** [reply]
Maybe I should change the value of `SW_QUERY_ZIPKIN_REST_HOST` as it should be the address of oap server rather than the address of UI pod?

**Reply by kezhenxu94** [reply]
Exactly, please change that address to the correct one

**Comment by Yangfisher1** [comment, marked as answer]
The problem is fixed by adding `SW_ZIPKIN_ADDRESS` environment  variable in [ui-deployment.yaml](https://github.com/apache/skywalking-kubernetes/blob/master/chart/skywalking/templates/ui-deployment.yaml).

``` yaml
        - name: SW_OAP_ADDRESS
          value: "http://{{ template "skywalking.oap.fullname" . }}:{{ .Values.oap.ports.rest }}"
        - name: SW_ZIPKIN_ADDRESS
          value: "http://{{ template "skywalking.oap.fullname" . }}:{{ .Values.oap.ports.zipkinui }}"
```

and configure the port for querying zipkin with ` --set oap.ports.zipkinui=9412`. Now it works perfectly.
